presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -428,14 +419,40 @@ private void invokeNoMoreSplitsIfNecessary()
             throw new PrestoException(NOT_SUPPORTED, "Original non-ACID files in transactional tables are not supported");
         }
 
-        // delta directories
+        fileIterators.addLast(
+                directory.getOriginalFiles().stream()
+                        .map(file -> file.getFileStatus())
+                        .map(fileStatus -> {
+                            try {
+                                return splitFactory.createInternalHiveSplit(fileStatus,
+                                        hdfsEnvironment.doAs(hdfsContext.getIdentity().getUser(), () -> fs.getFileBlockLocations(fileStatus.getPath(), 0, fileStatus.getLen())));
+                            }
+                            catch (IOException e) {
+                                throw new PrestoException(HIVE_BAD_DATA, "Could not fetch BlockLocations for original file " + fileStatus.getPath(), e);
+                            }
+                        })
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .iterator());
+
+        // First create Delete Deltas registry
+        DeleteDeltaLocations.Builder deleteDeltaLocationsBuilder = DeleteDeltaLocations.builder(path);
         for (AcidUtils.ParsedDelta delta : directory.getCurrentDirectories()) {
-            fileIterators.addLast(createInternalHiveSplitIterator(delta.getPath(), fs, splitFactory, splittable));
+            if (delta.isDeleteDelta()) {
+                deleteDeltaLocationsBuilder.addDeleteDelta(delta.getPath(), delta.getMinWriteId(), delta.getMaxWriteId(), delta.getStatementId());
```

Comment:
Maybe just
```
deleteDeltaLocationsBuilder.addDeleteDelta(delta);
```

Reply:
This way keeps tests simple, otherwise tests will have to create AcidUtils.ParsedDelta and FileStatus objects